controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -105,9 +105,18 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, err
 	}
 
-	if status.Nodes <= *kcp.Spec.Replicas {
-		// scaleUp ensures that we don't continue scaling up while waiting for Machines to have NodeRefs
-		return r.scaleUpControlPlane(ctx, cluster, kcp, controlPlane)
+	switch kcp.Spec.RolloutStrategy.Type {
```

Comment:
nit: Even if this should never happen due to validation/defaulting webhooks, I suggest testing for RolloutStrategy and RollingUpdate not nil, and erroring out in case they are nil.
(Might be probably better to do this as a first thing in the func)